CONTRIBUTING.md

```diff
@@ -1,57 +1,108 @@
+     - Rebase your branch onto the latest `dev` branch.
+     - Squash commits to streamline the history. Multiple commits can remain if they represent logical steps.
 
-### How to Use `/* prettier-ignore */`
+4. **Code Compliance:**
+   - Ensure your code adheres to the project’s formatting and linting guidelines.
 
-To use `/* prettier-ignore */`, simply place the comment on the line before the code you want to ignore. Here is an example:
+5. **Testing:**
+   - Include test coverage and ensure your branch passes all required tests and CI/CD pipelines.
 
-#### Example
+6. **Branch Updates:**
+   - Regularly update your branch with the latest changes from `dev` using `--rebase` to avoid conflicts.
 
-```javascript
-/* prettier-ignore */
-if (ShardeumFlags.VerboseLogs) console.log('Lengths of blocks after pruning', Object.keys(blocksByHash).length, Object.keys(readableBlocks).length)
-```
+7. **Integration Guidelines:**
+   - Use **rebase** instead of merge to integrate changes from `dev` and maintain a clean history.
+   - Direct submissions to the `main` branch are **prohibited**. All changes must go through `dev` first.
+
+8. **PR Reviews:**
+   - Once your PR is ready, undraft it and request reviews.
+   - Be receptive to feedback and make necessary adjustments promptly.
+
+---
+
+## Standardized Formatting Guidelines
+
+To ensure code consistency and maintainability, all contributors must adhere to the following formatting standards:
+
+1. **Prettier Configuration:**
+   - Use **Google TypeScript Prettier settings** to ensure a uniform style for all TypeScript files.
+
+2. **Minimal Use of Prettier Ignores:**
+   - Use `/* prettier-ignore */` sparingly to avoid inconsistent code styles across the project.
+
+3. **Linting:**
+   - Run the project's linter before submitting a pull request.
+   - Ensure your code complies with all linting rules defined in the project configuration.
+
+4. **Code Readability:**
+   - Write clear, readable code:
+     - Use meaningful variable and function names.
+     - Include comments where necessary to explain the purpose of code blocks.
+
+5. **Project Conventions:**
+   - Adhere to additional formatting and coding standards documented in the repository.
+
+6. **Commit Messages:**
+   - Follow a conventional commit message format for clarity and consistency. Refer to the [Commit Convention Guide](https://www.conventionalcommits.org/en/v1.0.0/).
+
+---
+
+## Additional Guidelines
+
+- **Coding Standards:**
+  - Follow the project's coding standards or style guide wherever possible.
+
+- **Testing:**
+  - Ensure all tests pass before submission, and provide adequate test coverage for your changes.
+
+- **Documentation:**
+  - Update documentation if your changes introduce new features or modify existing functionality.
+
+- **Community & Review Process:**
+  - Be respectful and receptive to feedback during the review process.
+  - Make necessary adjustments promptly based on reviewer suggestions.
+
+---
```

Comment:
We don't need this section, let's remove the `Additional Guideline` part, it is repeating most of the guidelines we already shared in the beginning